stdlib/public/runtime/Casting.cpp

```diff
@@ -2399,6 +2399,8 @@ namespace {
   };
 }
 
+static void installCallbacksToInspectDylib();
```

Comment:
Interesting, we've had reports of similar deadlocks with the code as it is today that might have the same underlying cause.